src/moonlink_connectors/src/pg_replicate/conversions/text.rs

```
@@ -150,7 +170,21 @@ impl TextFormatConverter {
             Type::JSON_ARRAY | Type::JSONB_ARRAY => Cell::Array(ArrayCell::Json(Vec::default())),
             Type::OID => Cell::U32(u32::default()),
             Type::OID_ARRAY => Cell::Array(ArrayCell::U32(Vec::default())),
-            _ => Cell::Null,
+            _ => match typ.kind() {
+                Kind::Composite(_) => Cell::Composite(Vec::default()),
+                Kind::Array(inner_type) => {
+                    // Handle arrays of composite types.
+                    // Note: inner_type here refers to the element type of the array.
+                    // PostgreSQL supports multi-dimensional arrays (e.g., text[][]),
+                    // but we currently only handle arrays of composite types here.
+                    match inner_type.kind() {
+                        Kind::Composite(_) => Cell::Array(ArrayCell::Composite(Vec::default())),
+                        Kind::Array(_) => Cell::Null, // TODO: Multi-dimensional arrays not yet handled
```

For unsupported types, you should fail instead of letting it pass silently.